src/uct/sm/mm/base/mm_iface.c

```diff
@@ -775,8 +775,8 @@ static UCS_CLASS_INIT_FUNC(uct_mm_iface_t, uct_md_h md, uct_worker_h worker,
     /* Allocate the receive FIFO */
     status = uct_iface_mem_alloc(&self->super.super.super,
                                  UCT_MM_GET_FIFO_SIZE(self),
-                                 UCT_MD_MEM_ACCESS_ALL, "mm_recv_fifo",
-                                 &self->recv_fifo_mem);
+                                 self->super.super.mem_access_flags,
```

Comment:
IMO the field mem_access_flags should be removed at all, we're not going to invalidate access to the shared memory FIFO